src/utils/ipfs.ts

```diff
@@ -0,0 +1,37 @@
+import { Ipfs } from './context'
+
+type PinataPinResponse = {
+  IpfsHash: string
+  PinSize: number
+  Timestamp: string
+}
+
+export async function pinJSONToIPFS(file: any, { uri, apiKey = '', apiSecret = '' }: Ipfs): Promise<PinataPinResponse> {
```

Comment:
This method appears to be generic but everything here is tailored for Pinata, maybe being specific about it helps any dev looking at it.